jsonnet/thanos-querier.libsonnet

```
@@ -266,8 +266,27 @@ function(params)
 
     deployment+: {
       spec+: {
+        strategy+: {
+          // Apply HA conventions
+          rollingUpdate: {
+            maxUnavailable: '25%',
+          },
+        },
         template+: {
           spec+: {
+            affinity+: {
+              podAntiAffinity+: {
```

```suggestion
              podAntiAffinity: {
```

We don't need to extend this, so it can be replaced

---

should we enhance github.com/thanos-io/kube-thanos to support the configuration of soft/hard affinity like we do in kube-prometheus?

---

Maybe we should create a library for such common functions/snippets? We already have a lib directory in kube-prometheus in https://github.com/prometheus-operator/kube-prometheus/tree/main/jsonnet/kube-prometheus/lib, which we can use for this. We also already have a function for creating anti-affinity rules [here](https://github.com/prometheus-operator/kube-prometheus/blob/1d2a0e275af97948667777739a18b24464480dc8/jsonnet/kube-prometheus/addons/anti-affinity.libsonnet#L17-L28), which we can use as a base for this. Later we can apply this to kube-thanos and CMO as well. WDYT? We would need to be careful not to recreate whole ksonnet though :D

cc @kakkoyun